component/block_cache/block_cache.go

```diff
@@ -135,7 +135,22 @@ func (bc *BlockCache) SetNextComponent(nc internal.Component) {
 func (bc *BlockCache) Start(ctx context.Context) error {
 	log.Trace("BlockCache::Start : Starting component %s", bc.Name())
 
+	log.Debug("BlockCache::Start : Creating block pool")
+	bc.blockPool = NewBlockPool(bc.blockSize, bc.memSize)
+	if bc.blockPool == nil {
+		log.Err("BlockCache::Start : failed to init block pool")
+		return fmt.Errorf("config error in %s [failed to init block pool]", bc.Name())
```

Comment:
This used to fail the mount earlier as it was part of `configure` but now it will allow mount to go successfully.

Reply:
Parent waits for a signal from the child if the mount is successful or not and then terminates.
https://github.com/Azure/azure-storage-fuse/blob/main/cmd/mount.go#L518-L528

If the blockpool creation failed in the Start() method in child, the mount will fail and it will return error, which will be checked in the pipeline start method
```
func (p *Pipeline) Start(ctx context.Context) (err error) {
	p.Create()

	for i := len(p.components) - 1; i >= 0; i-- {
		if err = p.components[i].Start(ctx); err != nil {
			return err
		}
	}

	return nil
}
```

So, the parent process will also fail the mount, returning an error which will be displayed on the user's console.